governance/default.go

```diff
@@ -608,17 +608,18 @@ func newGovernanceCache() common.Cache {
 
 func (g *Governance) initializeCache() error {
 	// get last n governance change block number
-	indices, err := g.db.ReadRecentGovernanceIdx(params.GovernanceCacheLimit)
+	indices, err := g.db.ReadRecentGovernanceIdx(params.GovernanceIdxCacheLimit)
 	if err != nil {
 		return ErrNotInitialized
 	}
 	g.idxCache = indices
 
 	// Put governance items into the itemCache
 	for _, v := range indices {
-		if num, data, err := g.ReadGovernance(v); err == nil {
-			g.itemCache.Add(getGovernanceCacheKey(num), data)
-			atomic.StoreUint64(&g.actualGovernanceBlock, num)
+		if data, err := g.db.ReadGovernance(v); err == nil {
```

Comment:
What is the reason of changing this line?

Reply:
@ehnuje `g.ReadGovernance` and `g.db.ReadGovernance` works differently. `g.db.ReadGovernance` read the governance information stored at the block `v`.  `g.ReadGovernance` calculates the block number which have to be applied at the block `v`.

So basic design was `idxCache` is to hold the block numbers and `itemCache` is to hold the governance information of matching block number stored in idxCache. But because of the calculation, there was mismatch between idxCache's block numbers and governance information stored in `itemCache`